flowmachine/flowmachine/features/location/redacted_total_events.py

```diff
@@ -0,0 +1,85 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from typing import List
+
+from flowmachine.core import Query
+from flowmachine.features import TotalLocationEvents
+
+
+class RedactedTotalEvents(Query):
```

Comment:
I'm wondering whether it would make sense to implement this class as joining the `TotalLocationEvents` query to a `RedactedUniqueSubscriberLocations` query using the location ID columns? That should avoid having to re-implement the `TotalLocationEvents` SQL here, and would mean that `RedactedLocationMetric` is used by all redacted queries.

Reply:
Ahh. Nice :)

Reply:
Although the tradeoff is that one then needs to copy over all the args to the unique subscriber counts query, which is a bummer.

Reply:
Yeah, that's not ideal. All the required arguments are attribute of the `TotalLocationEvents` query, so wouldn't be a problem to implement, but it's a little messy.

Alternatively, `RedactedTotalLocationEvents` could take the same args as `TotalLocationEvents` and construct both queries internally, but that's inconsistent with the other `Redacted*` queries and seems like a step in the opposite direction to several recent flowmachine changes.

Reply:
I think on balance, I prefer the modified SQL really - in reality, this is likely to be necessary in a bunch of cases _anyway_.